### PR TITLE
Add a section to the docs on how to debug tasks.

### PR DESCRIPTION
**Pull request by dlorenc**
# Changes

I added a step I use frequently to inspect the state of the workspace before
and after my Task steps execute. Feel free to add more!

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [X] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

### CODE REVIEW

File: docs/tasks.md [hunk 1]
```diff
@@ -684,3 +685,46 @@ Except as otherwise noted, the content of this page is licensed under the
...
+For the most part, these all live somewhere under `/workspace`, but the exact layout can
+be tricky.
+To see where things are before your task runs, you can add a step like this:
+
+```yaml
+- name: build-and-push-1
+  image: ubuntu
+  command:
+  - /bin/bash
+  args:
```
Comment (ImJasonH) [inline docs/tasks.md]: Do you want to do this with `script:`? Or will that maybe cause even more confusion if you're trying to _debug_ `script:` ?

Comment (vdemeester) [reply docs/tasks.md]: That might be confusing indeed :angel:

File: docs/tasks.md [hunk 2]
```diff
@@ -684,3 +685,46 @@ Except as otherwise noted, the content of this page is licensed under the
 [Creative Commons Attribution 4.0 License](https://creativecommons.org/licenses/by/4.0/),
 and code samples are licensed under the
 [Apache 2.0 License](https://www.apache.org/licenses/LICENSE-2.0).
+
+## Debugging a Task
```
Comment (bobcatfish) [inline docs/tasks.md]: can you put a link to this in the TOC somewhere at the top of the file? might even be good to link from teh main README

File: docs/tasks.md [hunk 3]
```diff
@@ -684,3 +685,46 @@ Except as otherwise noted, the content of this page is licensed under the
...
+
+In software, we do things not because they are easy, but because we think they will be.
```
Comment (bobcatfish) [inline docs/tasks.md]: XD

File: docs/tasks.md [hunk 4]
```diff
@@ -684,3 +685,46 @@ Except as otherwise noted, the content of this page is licensed under the
...
+Lots of things can go wrong when writing a Task.
+This section contains some tips on how to debug one.
+
+### Inspecting the Filesystem
+
+One common problem when writing Tasks is not understanding where files are on disk.
+For the most part, these all live somewhere under `/workspace`, but the exact layout can
```
Comment (bobcatfish) [inline docs/tasks.md]: maybe we should link to some docs about /workspace here? (i think we have some) - also will need to update when jason's /tekton change goes in 😅

File: docs/tasks.md [hunk 5]
```diff
@@ -684,3 +685,46 @@ Except as otherwise noted, the content of this page is licensed under the
...
+- name: build-and-push-1
+  image: ubuntu
+  command:
+  - /bin/bash
+  args:
+  - -c
+  - |
+    set -ex
+    find /workspace
+```
```
Comment (bobcatfish) [inline docs/tasks.md]: oh yeah good call, i do this all the time too - to the point where maybe we should consider adding a "debug" mode that does something like this?

**Pull request merged**